csharp/src/Google.Protobuf/Collections/MapField.cs

```diff
@@ -170,15 +169,14 @@ public bool Remove(TKey key)
         /// <returns><c>true</c> if the map contains an element with the specified key; otherwise, <c>false</c>.</returns>
         public bool TryGetValue(TKey key, out TValue value)
         {
-            LinkedListNode<KeyValuePair<TKey, TValue>> node;
-            if (map.TryGetValue(key, out node))
+            if (map.TryGetValue(key, out LinkedListNode<KeyValuePair<TKey, TValue>>? node))
             {
                 value = node.Value.Value;
                 return true;
             }
             else
             {
-                value = default(TValue);
+                value = default(TValue)!;
```

Comment:
Sure. Was trying to avoid a large number of C#4+-isms in this PR to reduce diffs. But per other comments I'll loosen that up in upcoming pushes. I still plan to do a follow-on wave of fixing newer-C# style and ReSharper recommendations in another PR.